backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -1558,4 +1578,219 @@ public FutureProjectDAO(
             unused -> updateModifiedTimestamp(request.getId(), new Date().getTime()), executor)
         .thenCompose(unused -> updateVersionNumber(request.getId()), executor);
   }
+
+  public InternalFuture<Project> createProject(CreateProject request) {
+    // Validate if current user has access to the entity or not
+    return checkProjectPermission(null, ModelDBActionEnum.ModelDBServiceActions.CREATE)
+        .thenCompose(unused -> createProjectHandler.convertCreateRequest(request), executor)
+        .thenCompose(
+            project ->
+                jdbi.withHandle(
+                        handle ->
+                            handle
+                                .createQuery(
+                                    "SELECT p.id From project p where p.name = :projectName AND p.deleted = :deleted")
+                                .bind("projectName", project.getName())
+                                .bind("deleted", true)
+                                .mapTo(String.class)
+                                .list())
+                    .thenCompose(
+                        deletedProjectIds ->
+                            deleteEntityResourcesWithServiceUser(
+                                deletedProjectIds,
+                                ModelDBResourceEnum.ModelDBServiceResourceTypes.PROJECT),
+                        executor)
+                    .thenApply(status -> project, executor),
+            executor)
+        .thenCompose(createProjectHandler::insertProject, executor)
+        .thenCompose(
```

Comment:
This lambda is way too nested. I can't follow what it's doing. Let's split this method into helper methods for clarity. I _think_ it's correct, but I'm not confident since I can't be sure I follow.

Reply:
Fix and try to split method in helper functions

Reply:
This is still somewhat heavily nested, but It is better than before.